layer/unittest/trace_event_log_tests.cc

```diff
@@ -294,8 +294,37 @@ class InvalidCompleteEvent : public Event {
   TraceEventAttr trace_event_;
 };
 
+// TraceEventLogger logs the events in the Trace Event format to the output
+// given in its constructor. There is no need to add '\n' at the end of the
+// input of the `AddEvent()` method. This is handled by the implementation. The
+// only valid methods after calling `EndLog()` is `EndLog()`.
+class TraceEventLogger : public EventLogger {
+ public:
+  TraceEventLogger(LogOutput *out) : out_(out) { assert(out); }
+
+  void AddEvent(Event *event) override {
+    std::string event_str = EventToTraceEventString(*event);
+    out_->LogLine(event_str);
+  }
+
+  // Writes '[', indicating the start point of the JSON array.
+  void StartLog() override { out_->LogLine("["); }
+
+  // The ']' at the end of the JSON array is optional in the Trace Event format.
+  // We exploit this to allow multiple layers write into the same file.
+  void EndLog() override {}
+
+  void Flush() override { out_->Flush(); }
+
+ private:
+  LogOutput *out_ = nullptr;
+};
+
+constexpr int64_t event_timestamp = 1401;
```

Comment:
Can you move this towards the top of the file and call it something like `kTestEventTimestamp`?